src/client/multiplayer/multiplayer.js

```diff
@@ -91,10 +91,7 @@ export class Multiplayer {
   connect() {
     if (!this.socket) {
       if (this.server) {
-        this.socket = io(
-          'http://' + this.server + '/' + this.gameName,
-          this.socketOpts
-        );
+        this.socket = io(this.server + '/' + this.gameName, this.socketOpts);
```

Comment:
You can also connect using '//hostname:port' and socket.io will choose the appropriate transport in order to avoid a mixed connection.

How about we leave the server specification as `<hostname>:<port>` and use this to determine when to switch to https. In order to support the use case of forcing SSL (even on http connections), socket.io has a `{ secure: true }` option that you can pass here that we could probably also expose in the multiplayer spec like this:

```
multiplayer: { server: 'hostname:port', secure: true },
```

Reply:
fine, not sure what to do about the tests though... 'socket.io' still seems to default to `about` protocol